### DISCUSSION TITLE
6.1.4 Not ingesting at all - Windows - SDRTrunk

### DISCUSSION THREAD

**Discussion by noamlivne** [chuot/rdio-scanner#108] [General]
Hi @chuot,
I installed 6.1.4 now, imported the Config from 6.1.3, but nothing seems to be ingested.
As if the program is not going to the Dirwatch.
Strange. 
AFAIK I did not do any change.
Windows10, SDRTrunk.
Thank you very much in advance.

**Comment by chuot** [comment]
Indeed, I just discovered a bug in the meta data parser. Working on a fix.

**Comment by chuot** [comment]
Two things to check:

1. Is your monitored folder exists prior to start Rdio Scanner?
2. Is you dirwatch definition is disabled? I wasn't checking this prior to v6.1.4. So if it is disabled, it will be ignored as of v6.1.4.

**Reply by noamlivne** [reply]
1. My monitored folder did exist prior to start Rdio-Scanner. 
2. It is not disabled. I used the Config from 6.1.3 that did ingest well. And just to make sure, I also started now 6.1.4 from scratch. In both cases the dirwatch is not disabled, and in both cases there was no ingestion.

**Comment by chuot** [comment]
v6.1.5 is just out, please give it a try and let me know. Thanks

**Reply by noamlivne** [reply]
Thanks a lot! I can't find it in the Release tab. Should I download it from somewhere else?

**Reply by chuot** [reply]
Yeah, there was a minor glitch with it. I just released v6.1.6.

**Reply by noamlivne** [reply]
Found 6.1.6 there... Will test. Thank again

**Reply by noamlivne** [reply]
Unfortunately, still no ingestion for me in 6.1.6

**Reply by chuot** [reply]
I tested dirwatch type sdrtrunk ingestion on windows 10 and it works for me.

Do your run Rdio Scanner as a service? Can you stop the service and run the rdio-scanner executable directly at the command prompt? Is there any strange error messages?

**Reply by chuot** [reply]
Oh, and remember dirwatch works only with local folders, not network mounted folders.

**Reply by noamlivne** [reply]
I am not sure what is "service".
I run the command I was running since a few versions before.

When I unzip the .zip file I download from the Releases tab, I run this in the cmd terminal
rdio-scanner -ssl_create
Then I run this as a .bat file:
rdio-scanner -ssl_cert_file server.crt -ssl_key_file server.key -ssl_listen :443

I do not see any error in the black "cmd" terminal for both the commands I issue.

I then open the Admin Dashboard and Import my config.json from 6.1.3 and Save.

The black cmd terminal indicates that there was a configuration change (and if I open the APP, a listener joins), but nothing is ingested from my Recordings directory, as opposed to what was the case with 6.1.3. All my Talkgroups appear in the Select TG screen fine. But nothing in the cmd, or the Live Feed or the Search Call buttons.
When trying to ctrl+z or ctrl+x or ctrl+c in the cmd terminal, nothing happens. The program does not seem to stop, but there is no error anyway.

**Reply by noamlivne** [reply]
The dirwatch's location did not change from 6.1.3.
Rdio-scanner is at C:something.../downloads
and my Recording location is at C:users/something

**Reply by chuot** [reply]
Are you keeping each versions in there own directory with separated data files (database and certificates)?

Actually, you should stop Rdio Scanner server, unzip the new version over the old one and restart the Rdio Scanner server.

You don't have to export and import the config, nor recreate the SSL certificates. The database gets updated automatically on newer version.

Can you erase the dirwatch definition from the admin dashboard and recreate it back?

**Reply by noamlivne** [reply]
Yep, I am keeping each version in their own directory for backup reasons + the Database file gets so huge even though I have the calls deleted automatically and the database pruned each day.
I will now backup the 6.1.3 that works for me and use it as the base for the new version 6.1.6.
If needed, I will create the dirwatch definition from scratch.

**Reply by noamlivne** [reply]
Unfortunately, both of your tips did not help.
Everything seems to work fine, except the dirwatch / ingestion part.
Maybe when others will download 6.1.6 they can chime-in.

**Reply by chuot** [reply]
Would you mind testing this attached version. It has a bunch of debug outputs for the dirwatch. Please grab all the lines with starting with DBEUG: and post them back.

[rdio-scanner-windows-amd64-v6.1.6-debug1.zip](https://github.com/chuot/rdio-scanner/files/7927473/rdio-scanner-windows-amd64-v6.1.6-debug1.zip)

**Reply by noamlivne** [reply]
I ran-over the "old" file with the debug-file.
Nothing worked, but for a second it had ingested one call(?). Perhaps one call that contains ALL the calls it did not ingest till now. But that is not good.
Anyway, here is the only debug I found:
DBEUG: dirwatch C:\Users\noaml\recordings stopping
I created the dirwatch from scratch, just in case, but no change.

**Reply by chuot** [reply]
Can you test again with this one?

[rdio-scanner-windows-amd64-v6.1.6-debug2.zip](https://github.com/chuot/rdio-scanner/files/7927753/rdio-scanner-windows-amd64-v6.1.6-debug2.zip)

**Reply by noamlivne** [reply]
These are the debugs this time, immediately when I started the program for:
DBEUG: dirwatches.read
DBEUG: dirwatches.start
DBEUG: dirwatch C:\Users\noaml\recordings starting

Note that this time, when I went to Delete the dirwatch, the rdio-scanner crashed with a 504 Gateway error. Happened twice, but not on the 3rd time, then I managed to create the dirwatch from scratch and got these debugs:
DBEUG: new config
DBEUG: dirwatches.write
DBEUG: dirwatch C:\Users\noaml\recordings stopping
DBEUG: dirwatches.read
DBEUG: dirwatches.start
2022/01/24 20:05:47 configuration changed
DBEUG: new config
DBEUG: dirwatches.write
DBEUG: dirwatches.read
DBEUG: dirwatches.start
DBEUG: dirwatch C:\Users\noaml\recordings starting
2022/01/24 20:07:10 configuration changed
But no ingestion

**Reply by chuot** [reply]
Can you post the full log when it crashed?

**Reply by noamlivne** [reply]
Now I ctrl-z / ctrl-x / ctrl-c and got this:

DBEUG: dirwatch C:\Users\noaml\recordings loop exited while still running, restarting
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x138e938]

goroutine 115 [running]:
github.com/fsnotify/fsnotify.(*Watcher).Close(0x16fe540)
        /home/chuot/go/pkg/mod/github.com/fsnotify/fsnotify@v1.5.1/windows.go:53 +0x18
main.(*Dirwatch).Start.func1()
        /home/chuot/Dev/rdio-scanner/server/dirwatch.go:518 +0x43a
created by main.(*Dirwatch).Start
        /home/chuot/Dev/rdio-scanner/server/dirwatch.go:530 +0x26f

**Reply by noamlivne** [reply]
When it crashed, the cmd terminal window closed immediately, so I did not manage to see anything, besides the 504 error in the browser, probably. I do not think there is a log file I can send you.

**Reply by chuot** [reply]
We are making progress! One last version to test.

[rdio-scanner-windows-amd64-v6.1.6-debug3.zip](https://github.com/chuot/rdio-scanner/files/7927970/rdio-scanner-windows-amd64-v6.1.6-debug3.zip)

**Reply by noamlivne** [reply]
This is what I get after starting the program and then Deleting the dirwatch and setting it up from scratch and Saving. But still no ingestion:

C:\Users\noaml\Downloads\rdio-scanner-windows-amd64-v6.1.3>rdio-scanner -ssl_cert_file server.crt -ssl_key_file server.key -ssl_listen :443

Rdio Scanner v6.1.6
----------------------------------
2022/01/24 20:33:48 server started
2022/01/24 20:33:48 base folder is C:\Users\noaml\Downloads\rdio-scanner-windows-amd64-v6.1.3
DBEUG: dirwatches.read
DBEUG: dirwatches.start
DBEUG: dirwatch C:\Users\noaml\recordings starting
2022/01/24 20:33:49 main interface at http://DESKTOP-38VLALU:3000
2022/01/24 20:33:49 main interface at https://DESKTOP-38VLALU
2022/01/24 20:33:49 admin interface at https://DESKTOP-38VLALU/admin
DBEUG: new config
DBEUG: dirwatch C:\Users\noaml\recordings loop exited while still running, restarting
DBEUG: dirwatches.write
DBEUG: dirwatches.read
DBEUG: dirwatches.start
2022/01/24 20:34:53 configuration changed
DBEUG: new config
DBEUG: dirwatches.write
DBEUG: dirwatches.read
DBEUG: dirwatches.start
DBEUG: dirwatch C:\Users\noaml\recordings starting
2022/01/24 20:35:50 configuration changed

And nothing happens when I do ctrl-z/x/c

**Reply by chuot** [reply]
You should see something like `2022/01/24 13:42:03 terminated` after doing a ctrl-c.

Also, what have you changed in the admin dashboard while doing the test? I see 2 `configuration changed`.

**Reply by noamlivne** [reply]
Unfortunately I did not see anything after ctrl-c, as if it was frozen.
The first change is Delete the dirwatch and Save.
The second change is create a fresh dirwatch and Save

**Reply by noamlivne** [reply]
I will have to disconnect now. Feel free to send me more (debug) versions to test. I hope I will get to it tomorrow.
Thank you very much for your efforts.

**Reply by noamlivne** [reply]
I tried from scratch on a different, more modern computer. Unfortunately no ingestion.
The only glimmer of hope I had was when I first installed the official 6.1.6 and set it to the "old" dirwatch I tested on this modern laptop in the past: it ingested old recordings of a month(?) ago. But that is it. It did not ingest anything new.  I changed the computer Date formats, I changed the computer's date back in time, I started a new dirwatch in the rdio-scanner main directory from scratch. 
Nothing. 
Oh well, I will stick with 6.1.3 till perhaps another person will have this issue.

**Reply by chuot** [reply]
I'm still on it. Whenever you have time, can you test this one?

[rdio-scanner-windows-amd64-v0.0.0-debug4.zip](https://github.com/chuot/rdio-scanner/files/7942915/rdio-scanner-windows-amd64-v0.0.0-debug4.zip)

**Reply by noamlivne** [reply]
Thank you very much for your efforts.
I installed it on top of the last 6.1.6Debug3, and also alone.
When installed on top of the 6.1.6Debug3, the panel showed in the Search Call window, the calls from two days ago, that were probably saved in the .db in that directory. No ingestion and no DEBUG indication in the Terminal.
When installed alone there were no (old) calls, not new and not ingested, and no DEBUG indication in the Terminal.

**Reply by chuot** [reply]
While debug4 is running, if you just manually copy a new MP3 file from SDRTrunk to the monitored folder, does it get ingested?

**Reply by noamlivne** [reply]
Nope. Not manually copying nor "live" helps. No ingestion.

**Reply by noamlivne** [reply]
I will have to disconnect now. Feel free to send me more (debug) versions to test. I hope I will get to it tomorrow.
If you want, you can start from 6.1.3 and step-by-step introduce the changes you made since then, to try and pin point the issue.
Thank you very much for your efforts.

**Reply by chuot** [reply]
Found it! You should be ok with v6.1.7.

**Reply by noamlivne** [reply]
WORKS! Thank you very much!
6.1.7 Started ingestion immediately.  
Great work!
I did have two issues, but they are probably not related:
1. After 3 minutes of ingestion it stopped. I do not know why. (Maybe a glitch from SDRTrunk). I restarted the program and all was great again.
2. And, I swiped Tailscale and your Android app from my smartphone and got this error in the program:
2022/01/27 19:02:25 client.conn.writemessage: write tcp 100.79.29.86:443->100.89.248.62:42762: wsasend: An existing connection was forcibly closed by the remote host.
But besides those two one-time issues, till now, it seems to ingest excellent.

If I will encounter issues similar to those above again, or new ones, I will open a new thread.
Thank you again.
